### PR TITLE
added profile module

### PR DESCRIPTION
**Pull request by MariiaPolova**

### CODE REVIEW

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 1]
```diff
@@ -6,15 +6,22 @@
...
+//	}
 
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		Request<T> requestObj = new Request<T>();
+		requestObj.setHeader(msg.getHeader());
+
+		//requestObj is created after Jackson mapper conversion
+		return doHandle(requestObj);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:22]: @MariiaPolova What about data field?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -6,15 +6,22 @@
...
 	}
 
+	public abstract Response<R> doHandle(Request<?> requestObj);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: @MariiaPolova What is the difference between this method and handle method ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 3]
```diff
@@ -2,17 +2,37 @@
...
-	@Override
-	public String getRouteKey(Request<?> msg) {
+	private Map<String, T> handlerMap = new HashMap();
+
+	@Autowired
+	private List<T> handlers;
+
+	@PostConstruct
+	public void init(){
+		for(T handler : handlers)
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java:21]: @MariiaPolova Add curly braces to for.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 4]
```diff
@@ -2,17 +2,37 @@
...
+
+	public String getName() {
 		return null;
 	}
 
-	@Override
 	public Response<?> handle(Request<?> msg) {
-		return null;
+		//String command = msg.getHeader().getCommand();
+		return handlerMap.get(getRouteKey(msg)).handle(msg);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java:31]: @MariiaPolova What will be if handler not found ?

File: softgroup/services/authorization/authorization-impl/main/java/com/softgroup/authorization/impl/router/AuthorizationRouter.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
  * Created by polev on 24.02.2017.
  */
 @Component
-public class HandleRouter<T extends RouterHandler> implements RouterHandler {
+public class AuthorizationRouter<T extends RouterHandler> implements RouterHandler {
 
     private Map<String, T> handlerMap;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/main/java/com/softgroup/authorization/impl/router/AuthorizationRouter.java:21]: @MariiaPolova Why we need this map here ?

File: softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestLoginHandler.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by polev on 25.02.2017.
+ */
+
+@Component
+public class AuthorizationRequestLoginHandler
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestLoginHandler.java:17]: @MariiaPolova No need to put Authorization to handler name. It's already in authorization module.
Better rename this to LoginRequestHandler

File: softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestLoginHandler.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        extends AbstractRequestHandler<LoginRequest, LoginResponse>
+        implements AuthorizationRequestHandler {
+    public String getRouteKey(Request<?> msg) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestLoginHandler.java:20]: @MariiaPolova Why we need this here ?

File: softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by polev on 25.02.2017.
+ */
+
+@Component
+public class AuthorizationRequestRegisterHandler
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java:17]: @MariiaPolova Rename

File: softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        extends AbstractRequestHandler<RegisterRequest, RegisterResponse>
+        implements AuthorizationRequestHandler {
+
+    public String getName() {
+        return "register";
+    }
+
+    public Response<RegisterResponse> doHandle(Request<?> requestObj) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java:25]: @MariiaPolova Why wild-card Request ?

**Comment by Dominys** [comment]
@MariiaPolova Why you chose develop as name of the branch? Use meaningful branch names.
Where is PR for changes that you already made in master ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 1 above)
Comment (MariiaPolova) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:22]: @Dominys after setHeader() should be next code?
requestObj.setData((T)msg.getData());

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 2 above)
Comment (MariiaPolova) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: @Dominys 
handle() parses object and after that doHandle() receives it. I suggest, it includes repeatable actions, when doHandle() will be implemented by other handlers individually

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java (hunk 4 above)
Comment (MariiaPolova) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java:31]: @Dominys So firstly I need to check if needed handler exists, if not - return null, correct?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 1 above)
Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:22]: @MariiaPolova Something like that

File: softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java (hunk 9 above)
Comment (MariiaPolova) [reply softgroup/services/authorization/authorization-impl/src/main/java/handler/AuthorizationRequestRegisterHandler.java:25]: @Dominys 
fixed. `doHandle(Request<RegisterRequest> requestObj)` now

**Comment by MariiaPolova** [comment]
@Dominys I did it wrong from the beginning :( 
So I will do new PR with all new files, which were added to initial template (including new comments and fixes).

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 2 above)
Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: @MariiaPolova I mean that they use the same generic types

**Pull request closed without merging**